release-23.1:     changefeedccl: Fix setting propagation flake

miretskiy opened the pull request:
Backport 1/2 commits from #110601.

/cc @cockroachdb/release

---

Setting changes are asynchronous, particularly in tenants.
Set the feature setting directly instead.

Fixes #110300

Release note: None
Release justification: Test only fix

Commit:
changefeedccl: Fix setting propagation flake

Setting changes are asynchronous, particularly in tenants.
Set the feature setting directly instead.

Fixes #110300

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/110635)
<!-- Reviewable:end -->

Commit:
changefeedccl: Timeout test if error does not happen soon

Add a timeout when expecting an error to occur in tests.

Informs #110300

Release note: None

Commit:
changefeedccl: Add a test utility to timeout when waiting for error

Add `ExpectErrWithTimeout` test utility which times out
if the error does not occur withing expected threshold.
This utility function differes from ExpectErrSucceedsSoon
in that it does not retry the call -- it merely expects
an error to be returned withing timeout threshold.

Informs #110300

Release note: None

jayshrivastava reviewed (approved):
Btw `pkg/testutils/fuzz/BUILD.bazel` is empty. We probably don't need to commit that.
